pkg/nuctl/command/redeploy.go

```diff
@@ -321,11 +321,18 @@ func (d *redeployCommandeer) isFunctionInTerminalState(ctx context.Context, func
 		return true, nil
 	}
 
+	// unhealthy is not an end-state and function can become ready, so continue waiting
+	if function.Status.State == functionconfig.FunctionStateUnhealthy {
+		d.rootCommandeer.loggerInstance.WarnWithCtx(ctx,
+			"Function is unhealthy",
+			"functionName", functionName,
+			"message", function.Status.Message)
+	}
+
```

Comment:
Not really needed because we log the state again afterwards, so it will just be spammy.